external/PATCH-01-STRATUM

```diff
@@ -2935,12 +2935,15 @@ index 9f4bb3a9..f07e4843 100644
 +        *flags, table_key.get(),
 +        table_data.get());
 +    if (tdi_status != TDI_SUCCESS) {
++        if (tdi_status == TDI_OBJECT_NOT_FOUND) {
++            return ::util::OkStatus();
```

Comment:
Why are we return ok here if object is not found?

Reply:
the sde code checks all the tables that are present in p4 file and returns the object if present, for supposed there are two tables e_fwd and i_fwd, but the user adds entry only for i_fwd then we see an error saying that:

[New Thread 0x7fffbeffd640 (LWP 921955)]
E20221024 04:11:40.887918 921955 bf_sde_wrapper.cc:607] StratumErrorSpace::ERR_INTERNAL: Error retreiving information from TDI
E20221024 04:11:40.887974 921955 bf_sde_wrapper.cc:3605] Return Error: GetAllEntries(real_session->tdirt_session_, *dev_tgt, table, &keys, &datums) failed with StratumErrorSpace::ERR_INTERNAL: Error retreiving information from TDI
E20221024 04:11:40.887987 921955 bfrt_table_manager.cc:578] Return Error: tdi_sde_interface_->GetAllTableEntries( device_, session, table_id, &keys, &datas) failed with StratumErrorSpace::ERR_INTERNAL: Error retreiving information from TDI
E20221024 04:11:40.887998 921955 bfrt_table_manager.cc:639] Return error: ReadAllTableEntries(session, wanted_table_entry, writer) failed with StratumErrorSpace::ERR_INTERNAL: Error retreiving information from TDI
E20221024 04:11:40.888090 921955 bfrt_table_manager.cc:639] StratumErrorSpace::ERR_INTERNAL: Error retreiving information from TDIFailed to read all table entries for request .
E20221024 04:11:40.888222 921955 bfrt_node.cc:372] StratumErrorSpace::ERR_AT_LEAST_ONE_OPER_FAILED: One or more read operations failed.
E20221024 04:11:40.888250 921955 p4_service.cc:348] Failed to read forwarding entries from node 1: One or more read operations failed.
E20221024 04:11:40.888484 921955 utils.cc:101] StratumErrorSpace::ERR_INTERNAL: Error when opening /var/log/stratum/p4_reads.pb.txt.
E20221024 04:11:40.888504 921955 p4_service.cc:267] Failed to log the read request: Error when opening /var/log/stratum/p4_reads.pb.txt.

Reply:
Add a comment explaining that sde iterates through all the tables, and if no entry is present for that table, TDI_OBJECT_NOT_FOUND is not returned in which case it's no operation for flow dump for that table

Reply:
this is an edge case for object not found, this is supposed to WAD from SDE and there are different return statements from sde for other issues